src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,151 @@
+        } else {
+            Node<T> current = head;
+            while (current.next != null) {
+                current = current.next;
+            }
+            current.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+
+        Node<T> newNode = new Node<>(value);
+        if (index == 0) {
+            newNode.next = head;
+            head = newNode;
+        } else {
+            Node<T> prevNode = getNode(index - 1);
+            newNode.next = prevNode.next;
+            prevNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isNotValidIndex(index);
+
+        Node<T> current = getNode(index);
+        return current.element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        isNotValidIndex(index);
+
+        Node<T> current = getNode(index);
+        T oldValue = current.element;
+        current.element = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+
+        Node<T> removedNode;
+        if (index == 0) {
+            removedNode = head;
+            head = head.next;
+        } else {
+            Node<T> prevNode = getNode(index - 1);
+            removedNode = prevNode.next;
+            prevNode.next = removedNode.next;
+        }
+        size--;
+        return removedNode.element;
     }
 
     @Override
     public boolean remove(T object) {
+        if (isEmpty()) {
+            return false;
+        }
+
+        Node<T> current = head;
+        Node<T> prev = null;
+
+        while (current != null) {
+            if (Objects.equals(current.element, object)) {
+                if (prev == null) {
+                    head = current.next;
+                } else {
+                    prev.next = current.next;
+                }
+                size--;
+                return true;
+            }
+            prev = current;
+            current = current.next;
+        }
+
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> getNode(int index) {
+        Node<T> current = head;
+        for (int i = 0; i < index; ++i) {
+            current = current.next;
+        }
+        return current;
+    }
+
+    private void isNotValidIndex(int index) {
```

Comment:
Change the naming, it is not a boolean to be named like that. Maybe indexValidation or checkIndex